src/com/trollworks/gcs/spell/SpellTechnique.java

```diff
@@ -0,0 +1,214 @@
+
+        SkillLevel skillLevel = Technique.calculateTechniqueLevel(character, name, specialization, categories, def, difficulty, points, limited, limitModifier);
+        if (skillLevel.mLevel != -1) {
+            int bonusLevels = 0;
+            StringBuilder toolTip = new StringBuilder(skillLevel.mToolTip);
+            bonusLevels += getSpellBonuses(character, ID_COLLEGE,      getCollege(),     categories, toolTip);
+            bonusLevels += getSpellBonuses(character, ID_POWER_SOURCE, getPowerSource(), categories, toolTip);
+            bonusLevels += getSpellBonuses(character, ID_NAME,         name,             categories, toolTip);
+            skillLevel.mLevel         += bonusLevels;
+            skillLevel.mRelativeLevel += bonusLevels;
+        }
+
+        if(mLevel == null || !mLevel.isSameLevelAs(skillLevel)) {
+            mLevel = skillLevel;
+            if(notify) {
+                notify(ID_LEVEL, this);
+            }
+        }
+    }
+
+    // Copied from Spell class
+    private static int getSpellBonuses(GURPSCharacter character, String id, String qualifier, Set<String> categories, StringBuilder toolTip) {
+        int level = character.getIntegerBonusFor(id, toolTip);
+        level += character.getIntegerBonusFor(id + '/' + qualifier.toLowerCase(), toolTip);
+        level += character.getSpellComparedIntegerBonusFor(id + '*', qualifier, categories, toolTip);
+        return level;
+    }
+
+    @Override
+    public boolean isEquivalentTo(Object obj) {
+        if (obj == this)
+            return true;
+        if (obj instanceof SpellTechnique) {
+            SpellTechnique that = (SpellTechnique) obj;
+            if (mSpellPrerequisiteCount != that.mSpellPrerequisiteCount)
+                return false;
+            return super.isEquivalentTo(obj);
+        }
+        return false;
+    }
+
+    @Override
+    public String getLocalizedName() {
+        return I18n.Text("Spell (Ritual Magic)");
+    }
+
+    @Override
+    public String getXMLTagName() {
+        return TAG_SPELL_TECHNIQUE;
+    }
+
+    @Override
+    public int getXMLTagVersion() {
+        return CURRENT_VERSION;
+    }
+
+    @Override
+    public String getRowType() {
+        return I18n.Text("Spell (Ritual Magic)");
+    }
+
+    @Override
+    protected void prepareForLoad(LoadState state) {
+        super.prepareForLoad(state);
+    }
+
+    @Override
+    public RowEditor<? extends ListRow> createEditor() {
+        return new SpellTechniqueEditor(this);
+    }
+
+    /**
+     * @param builder The {@link StringBuilder} to append this technique's satisfied/unsatisfied
+     *                description to. May be {@code null}.
+     * @param prefix  The prefix to add to each line appended to the builder.
+     * @return {@code true} if this technique has its default satisfied.
+     */
+    public boolean satisfied(StringBuilder builder, String prefix) {
+        if(getCollege() == null || getCollege().isBlank()) {
```

Comment:
Please move the common call to `getCollege()` out to a variable so that it only gets called once